src/main/java/net/juniper/netconf/Device.java

```diff
@@ -165,20 +167,11 @@ public Device(JSch sshClient,
      * @return the hello RPC that represents those capabilities.
      */
     private String createHelloRPC(List<String> capabilities) {
```

Comment:
This is cool. I wonder if it could have been part of the Hello.java?

Reply:
It was an existing method - I just updated it to use the new Hello.java. 

I'd be reluctant to add the `]]>]]>` message separator as part of the `Hello.java` - as that class is solely dedicated to parsing/creating XML which the separator isn't (it's purely for NETCONF over SSH).

The remaining code, 
```java
Hello.builder()
     .capabilities(capabilities)
     .build()
     .toXML()
```
could be a static method of `Hello.java` (perhaps `Hello.createXmlOnlyWithCapabilities(capabilities)` ?) but am not really sure that's adding anything?

(there's another question in my mind if `Device.java` should be doing this, anyway? Surely it's NetconfSession that controls the details of the open session, and should therefore be in control of the shape of the `<hello>` that is sent. Especially if there was ever a desire to support the chunked framing mechanism in NETCONF `urn:ietf:params:netconf:base:1.1` - but I think that's definitely a discussion for another day!)

Reply:
👍